components/ws-manager/pkg/manager/monitor.go

```diff
@@ -1039,6 +1040,34 @@ func (m *Monitor) finalizeWorkspaceContent(ctx context.Context, wso *workspaceOb
 		}
 	}
 
+	var snc wsdaemon.WorkspaceContentServiceClient
+	var wsDaemonIsReachable = func(ctx context.Context, wso workspaceObjects) wait.ConditionWithContextFunc {
+		return func(ctx context.Context) (bool, error) {
+			snc, err = m.manager.connectToWorkspaceDaemon(ctx, wso)
+			if err != nil {
+				return false, err
+			}
+
+			return true, nil
+		}
+	}
+
+	err = wait.PollImmediateWithContext(ctx, 100*time.Millisecond, 5*time.Minute, wsDaemonIsReachable(ctx, *wso))
```

Comment:
Why `5*time.Minute`, what does it represent / guard against? I assume the max amount of time we might be willing to consider a node to be dead, say for example it is overwhelmed and not responding in a timely manner.

Reply:
> Why 5*time.Minute, what does it represent / guard against? I

Random restarts of ws-daemon.